### ISSUE TITLE
Derived properties with .collection and/or .parent dependency

### ISSUE THREAD

**Issue by ConneXNL** [AmpersandJS/ampersand-state#44]
I'm currently trying to use the following example.
```javascript   
   derived{
         picked: {
                deps: ['collection.parent'],
                fn: function () {
                    return this.collection.parent.isPicked(this);
                },
                cache: false
            },
     }
```
However, it does not trigger any event on this object that "picked" has changed. Is this intended? Is there a way to let this work?

As a workaround i removed the derived and changed it into:
```javascript  
{
        session:{
            picked: ['boolean', true, false],
        },

        initialize : {
            this.listenTo(this.collection.parent, "change:picked", function() {
                 this.picked = this.collection.parent.isPicked(this);
        });
}
```

It seems i'm also having an issue listening to a collection with a derived property.
```javascript  
        collections: {
            options: OptionCollection
        },
        derived: {
            top: {
                deps: ['options'],
                fn: function () {
                    console.log("running derived options");
                    return this.getTop();
                }
            }
        }
```

**Comment by HenrikJoreteg** [comment]
This has been sitting untouched for a while, and what it comes down to is that I think it's too easy to make a really hard to follow set of derived properties. Supporting children already added a decent amount of complexity, going both introduces even more potential for issues with circular dependencies, etc. 

Going to close this one.
